canisters/management.ts

```diff
@@ -124,6 +124,32 @@ export type ProvisionalTopUpCanisterArgs = {
     amount: nat;
 };
 
+export type HttpRequestArgs = {
+    url: string;
+    max_response_bytes: Opt<nat64>;
+    http_method: HttpMethod;
+    headers: HttpHeader[];
+    body: Opt<blob>;
+    transform_method_name: Opt<string>;
+};
+
+export type HttpMethod = Variant<{
+    GET: null;
+    HEAD: null;
+    POST: null;
+}>;
```

Comment:
Consider exporting a const of the same name. i.e.

```ts
export const HttpMethod = {
    GET: { GET: null },
    HEAD: { HEAD: null },
    POST: { POST: null }
};
```
_(Not sure of on the casing here, they might just be `Get`, `Head`, `Post`, but something like this)_

Then when you're making a call you can do:

```ts
yield ManagementCanister.http_request({
    url: `https://xkcd.com/642/info.0.json`,
    http_method: HttpMethod.GET,
    ...
}).with_cycles(cycle_cost_total);
```